src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -66,22 +76,73 @@
         return getAcceptableTokens();
     }
 
+    /**
+     * Whether to allow default keyword not in last but surrounded with case.
+     * @param newValue whether to ignore checking.
+     */
+    public void setSkipIfLastAndSharedWithCase(boolean newValue) {
+        skipIfLastAndSharedWithCase = newValue;
+    }
+
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST defaultGroupAST = ast.getParent();
         //default keywords used in annotations too - not what we're
         //interested in
         if (defaultGroupAST.getType() != TokenTypes.ANNOTATION_FIELD_DEF
                 && defaultGroupAST.getType() != TokenTypes.MODIFIERS) {
-            final DetailAST switchAST = defaultGroupAST.getParent();
-            final DetailAST lastGroupAST =
-                switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
-                log(ast, MSG_KEY);
+            if (skipIfLastAndSharedWithCase && containsCaseAfter(ast)) {
+                log(ast, MSG_KEY_SKIP_IF_LAST_AND_SHARED_WITH_CASE);
+            }
+            else {
+                boolean containsInLastGroup = ast.getPreviousSibling() == null;
+                if (!skipIfLastAndSharedWithCase
```

Comment:
another way of writing it would be:
````
            if (skipIfLastAndSharedWithCase && containsCaseAfter(ast)) {
                log(ast, MSG_KEY_SKIP_IF_LAST_AND_SHARED_WITH_CASE);
            }
            else {
                if ((!skipIfLastAndSharedWithCase || ast.getPreviousSibling() == null)
                        && isInLastCaseGroup(ast.getParent())) {
                    log(ast, MSG_KEY);
                }
            }
````
but it would still have the `!skipIfLastAndSharedWithCase`.